project.yaml

```diff
@@ -1501,18 +1501,61 @@ Layer:
         (SELECT
             way,
             way_area/NULLIF(!pixel_width!::real*!pixel_height!::real,0) AS way_pixels,
-            name,
-            ref,
-            admin_level
+            name
           FROM planet_osm_polygon
           WHERE boundary = 'administrative'
-            AND admin_level IN ('2', '4')
+            AND admin_level = '2'
             AND name IS NOT NULL
-            AND way_area > 750*!pixel_width!::real*!pixel_height!::real -- only labels for larger areas are displayed
-          ORDER BY admin_level ASC, way_area DESC
-        ) AS placenames_large
+            AND way_area > 100*!pixel_width!::real*!pixel_height!::real
+          ORDER BY way_area DESC
+        ) AS country_names
     properties:
       minzoom: 2
+  - id: "capital-names"
+    name: "capital-names"
+    class: ""
+    geometry: "point"
+    <<: *extents
+    Datasource:
+      <<: *osm2pgsql
+      table: |-
+        (SELECT
+            way,
+            name,
+            CASE
+              WHEN population ~ '^-?\d+(\.\d+)?$' THEN population::NUMERIC ELSE 0
```

Comment:
I would suggest to use the following code instead:

```
WHEN (population ~ '^[0-9]{1,8}$') THEN population::INTEGER
ELSE 0
```

Reasons:

- Tagging: Negative population values and non-integer floating point population count are not usefull values for the population tag. I think they should not be considered.
- Security: The digit limit `{1,8}` avoids SQL errors for extremely long (wrong) values like “54651654564564564564564564564564179875464661646”. Indeed this is less an issue with your code because you use NUMERIC and not INTEGER, but I would feel more comfortable when the digit count is limited.
- Consistency: This is the way it is done in placenames-medium